[WIP] WalletDetailsScreen, Keychain,TouchID(FaceID), bioLabels, LoginScreen, listOfPreviousAccounts, loginLogic / CM-278,156,270,279,275,276,277

e-molochko opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ozalexo/chronomint-rn/33)
<!-- Reviewable:end -->

ozalexo committed:
Pathed randomBytes function to make it optionally synchronous

Now iOS and Android have different shims (Android required Uint8Array.prototype.fill polyfill to launch)

e-molochko committed:
EncryptWallet changes

ozalexo commented on src/screens/Login/Start/StartContainer.js:
```diff
@@ -5,9 +5,14 @@
 import Start from './Start'
 
+
+const ACCESS_CONTROL_OPTIONS = ['None', 'Passcode', 'Password'];
```
;

ozalexo commented on src/screens/Login/Start/StartContainer.js:
```diff
@@ -5,9 +5,14 @@
 
+
+const ACCESS_CONTROL_OPTIONS = ['None', 'Passcode', 'Password'];
+const ACCESS_CONTROL_MAP = [null, Keychain.ACCESS_CONTROL.DEVICE_PASSCODE, Keychain.ACCESS_CONTROL.APPLICATION_PASSWORD, Keychain.ACCESS_CONTROL.BIOMETRY_CURRENT_SET]
```
Please format these array as column

ozalexo commented on src/screens/Login/Start/StartContainer.js:
```diff
@@ -30,13 +35,66 @@ class StartContainer extends PureComponent {
+    this.getBio()
+  }
+
+  getBio = () => {
```
It is OK for now. But in the future we will ned to call it once on app's first launch and keep it somewhere in the package/session

e-molochko committed:
Added encrypt/decrypt wallet opportunity

e-molochko committed:
Patch + temp changes

e-molochko committed:
Added react-native-touch-id

e-molochko committed:
Working on implementation of TouchId with react-native-keychain

e-molochko committed:
Added Login Screen
- Added login on password/touchId
- Added createAccount thunk
- Added login thunk

e-molochko committed:
Fixes

e-molochko committed:
Added switch between accountsList and createForm on StartScreen

ozalexo commented on packages/ethereum/redux/thunks.js:
```diff
@@ -2,3 +2,31 @@
+    dispatch(savePrivateKey(decryptedWallet.privateKey))
+
+    dispatch(bitcoinCreateWallet(ethAddress, bitcoinAddress))
+    dispatch(login(ethAddress))
```
This call is valid only for testing purposes.

e-molochko replied:
I am not sure, i think we need this call in create thunk, coz we need to pass current ethAddress

ozalexo commented on src/screens/Login/Start/StartContainer.js:
```diff
@@ -16,14 +17,15 @@ const mapStateToProps = (state) => {
+  constructor (props) {
+    super(props)
+    this.state={
+      showAccountsList: props.accounts && props.accounts.length !== 0 ? true : false,
```
Do we need to reset state on component mount/unmount?

e-molochko replied:
I think we dont have to. Because user can accidentally go to other page and  then go back. For now he will have data saved

ozalexo commented on src/screens/Login/Login/LoginContainer.js:
```diff
@@ -67,16 +67,14 @@ class LoginContainer extends PureComponent {
 
   authenticate = () => {
-    return TouchID.authenticate()
+    return TouchID.authenticate('ChronoWallet Application')
```
May we use a constant here instead of 'ChronoWallet Application'?

e-molochko replied:
Yeah i will change. It was for testing purposes. Should we add it to a constant or add it to i18n?

ozalexo commented on src/screens/Login/Login/LoginContainer.js:
```diff
@@ -85,7 +85,7 @@ class LoginContainer extends PureComponent {
     } = this.props.navigation.state.params
     const credentials = await Keychain.getInternetCredentials(account.address)
     if (credentials.password === password) {
```
To check that password is correct we need to encrypt a wallet with this password. User will be logged in in case of success decryption only.

e-molochko committed:
Fixed password verification on Login screen

e-molochko committed:
Added biometryType label

e-molochko committed:
Some fixes

e-molochko committed:
Added screen with WalletInfo after click on it in WalletList

e-molochko committed:
Fixes

e-molochko committed:
Patch packages

e-molochko committed:
Added Modals
- Send => EnterPassword => ConfirmSend

ozalexo committed:
Fixed Android non-debug mode:

Error "Super expression must either be null or function"

e-molochko committed:
Little fixes + comments changes

ozalexo commented:
All these changes were merged in scope of https://github.com/ozalexo/ChronoMint-RN/pull/34
